### PR TITLE
Nasu

### PR DESCRIPTION
**Pull request by yiancar**
Add Nasu keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/nasu/info.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Nasu",
+    "url": "https://yiancar-designs.com",
+    "maintainer": "Yiancar-Designs",
+    "width": 17.85,
+    "height": 8.9,
+    "layouts": {
+        "LAYOUT": {
```
Comment (zvecr) [inline keyboards/nasu/info.json:8]: Does not correctly key to nasu.h content
```suggestion
        "LAYOUT_alice_split_bs": {
```

File: keyboards/nasu/rules.mk [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# MCU name
+MCU = STM32F072
+
+# Bootloader selection
+BOOTLOADER = stm32-dfu
+
+# Wildcard to allow APM32 MCU
+DFU_SUFFIX_ARGS = -v FFFF -p FFFF
+
```
Comment (zvecr) [inline keyboards/nasu/rules.mk:9]: Copy-Paste or actual intentions to use APM32?

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/nasu/rules.mk (hunk 2 above)
Comment (yiancar) [reply keyboards/nasu/rules.mk:9]: Intention to find whatever is on the market :)

File: keyboards/nasu/rules.mk [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/nasu/rules.mk:25]:
```suggestion
```

**Review by drashna — APPROVED** [review]

File: keyboards/nasu/rules.mk [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+# Build Options
+#   comment out to disable the options.
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/nasu/rules.mk:23]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/nasu/rules.mk [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# Wildcard to allow APM32 MCU
+DFU_SUFFIX_ARGS = -v FFFF -p FFFF
+
+# Do not put the microcontroller into power saving mode
+# when we get USB suspend event. We want it to keep updating
+# backlight effects.
+OPT_DEFS += -DCORTEX_ENABLE_WFI_IDLE=TRUE
+
+# Build Options
+#   comment out to disable the options.
```
Comment (fauxpark) [inline keyboards/nasu/rules.mk:16]:
```suggestion
#   change yes to no to disable
```

File: keyboards/nasu/info.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Nasu",
+    "url": "https://yiancar-designs.com",
+    "maintainer": "Yiancar-Designs",
+    "width": 17.85,
+    "height": 8.9,
```
Comment (fauxpark) [inline keyboards/nasu/info.json:6]:
```suggestion
```

File: keyboards/nasu/info.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (fauxpark) [inline keyboards/nasu/info.json:1]: Needs to be redone without rotation.
image

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/nasu/info.json (hunk 7 above)
Comment (yiancar) [reply keyboards/nasu/info.json:1]: Fixed

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**